Detectors/TRD/simulation/src/Trap2CRU.cxx

```diff
@@ -668,28 +687,39 @@ void Trap2CRU::convertTrapData(o2::trd::TriggerRecord const& triggerrecord, cons
       uint32_t crudatasize = 0; // in 256 bit words.
       //loop over tracklets for mcm's that match
       //we have some data on this link
+      LOG(info) << __func__ << " " << __LINE__;
       int trackletcounter = 0;
+      bool firstTrackletOfLink = true;
       // if (mCurrentTracklet < endtrackletindex && mCurrentTracklet >= starttrackletindex && mCurrentDigit >= startdigitindex && mCurrentDigit < enddigitindex) {
       //if (mCurrentTracklet < endtrackletindex && mCurrentTracklet>=starttrackletindex mTracklets.size() || mCurrentDigit < mDigits.size()) {
       while (isTrackletOnLink(linkid, mCurrentTracklet) && mCurrentTracklet < endtrackletindex) { //}&& triggerrecord.getNumberOfTracklets()>0) {
+        LOG(info) << __func__ << " " << __LINE__;
         // still on an mcm on this link
+        if (firstTrackletOfLink) {
```

Comment:
This looks like the tracklet HCH is only written if there are tracklets in the event. The hardware will always send it.

Reply:
3 lines down from there is that question that you answer above. Fixing.

Reply:
done